org.eclipse.jdt.ls.core/gradle/android/init.gradle

```diff
@@ -0,0 +1,430 @@
+            return
+        }
+        if (!project.hasProperty(ECLIPSE_PROPERTY)) {
+            return
+        }
+        File androidSDKFile = getAndroidSDKFile(project)
+        if (androidSDKFile == null) {
+            return
+        }
+        project.afterEvaluate {
+            EclipseModel eclipseModel = (EclipseModel) project.property(ECLIPSE_PROPERTY)
+            // eclipseModel.synchronizationTasks() is available since Gradle 5.4
+            // see: https://www.eclipse.org/community/eclipse_newsletter/2019/june/buildship.php
+            if (GradleVersion.version(project.getGradle().getGradleVersion()) >= GradleVersion.version("5.4")) {
+                List<Object> synchronizationTasks = new ArrayList<>()
+                for (Object variant : getAndroidDebuggableVariants(project)) {
+                    try {
+                        synchronizationTasks.add(variant.properties.get("variantData").properties.get("taskContainer").properties.get("compileTask"))
+                    } catch (NullPointerException ignored) {
+                        // NPE occurs when the variant doesn't have related properties, we just skip these unsupported scenarios
+                    }
+                }
+                eclipseModel.synchronizationTasks(synchronizationTasks)
+            }
+            // Get project configurations from supported sourceSets names and configuration keys,
+            // and add them to eclipseModel.classpath.plusConfigurations
+            addPlusConfiguration(project, eclipseModel)
+            eclipseModel.classpath.setDownloadSources(true)
+            // Remove JDK container since android project should use embedded JDK types included in android.jar
+            eclipseModel.classpath.containers.removeIf(container -> container.contains("JavaSE"))
+            // Add supported source sets to source folders of eclipse model
+            eclipseModel.classpath.file.whenMerged(new AddSourceFoldersAction(project))
+            // Add data binding files to source folders of eclipse model
+            eclipseModel.classpath.file.whenMerged(new AddDataBindingFilesAction(project))
+            // Add R files to source folders of eclipse model
+            eclipseModel.classpath.file.whenMerged(new AddRFilesAction(project))
+            // Add buildconfig files to source folders of eclipse model
+            eclipseModel.classpath.file.whenMerged(new AddBuildConfigFilesAction(project))
+            // Add android.jar to project classpath of eclipse model
+            eclipseModel.classpath.file.whenMerged(new AddAndroidSDKAction(androidSDKFile))
+            // Add project dependencies to project classpath of eclipse model
+            // for aar dependencies, extract classes.jar and add them to project classpath
+            eclipseModel.classpath.file.whenMerged(new AddProjectDependenciesAction(project))
+        }
+    }
+
+    private static boolean isSupportedAndroidProject(Project project) {
```

Comment:
The method name confused me a little bit. Why project having `ANDROID_KOTLIN_PLUGIN_ID` is considered as not supported?

Reply:
since kotlin plugin will introduce some source folders like `src/main/kotlin` and related compile tasks, it's mainly used in Java/Kotlin polyglot project which is not supported in jdtls yet. So I temporary filter it here.

Reply:
I see. Let's add JavaDoc for this method for explanation